src/app/components/Fleet/FleetDetailsContent.tsx

```diff
@@ -61,101 +70,162 @@ const FleetStatus = ({ status }: { status: FleetStatus }) => {
   );
 };
 
+const FleetDevicesTab = ({
+  fleetName,
+  onDevicesLoaded,
+}: {
+  fleetName: string;
+  onDevicesLoaded: (deviceCount: number) => void;
+}) => {
+  const [deviceList, loadingDevices, devicesError] = useFetchPeriodically<DeviceList>({
+    endpoint: `devices?owner=${fleetName}`,
+  });
+
+  React.useEffect(() => {
+    onDevicesLoaded(deviceList?.items?.length || 0);
+  }, [deviceList, onDevicesLoaded]);
+
+  let tabContent;
+  if (loadingDevices || deviceList === undefined) {
+    tabContent = <Spinner />;
+  }
+  if (deviceList?.items) {
+    tabContent =
+      deviceList.items.length === 0 ? (
+        <EmptyState>
+          <EmptyStateBody>The fleet has no associated devices</EmptyStateBody>
+        </EmptyState>
+      ) : (
+        <DeviceTable devices={deviceList.items} showFleet={false} />
+      );
+  }
+
+  return (
+    <>
+      {devicesError && <Alert title="Error loading the fleet devices" />}
```

Comment:
Lets visualize the error too
```suggestion
      {devicesError && <Alert title="Error loading the fleet devices">{devicesError}</Alert>}
```